# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/linux/hotfix-mips64-sgi-ip32-r10k.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Apply ip28 indigo 2 r10k speculative load-store workaround for r10k/r120 o2 ip32, too.

--- linux-6.6/arch/mips/Kconfig.vanilla	2024-01-06 20:52:06.032449685 +0100
+++ linux-6.6/arch/mips/Kconfig	2024-01-06 20:52:47.309454246 +0100
@@ -785,7 +785,7 @@
 	select R5000_CPU_SCACHE
 	select RM7000_CPU_SCACHE
 	select SYS_HAS_CPU_R5000
-	select SYS_HAS_CPU_R10000 if BROKEN
+	select SYS_HAS_CPU_R10000
 	select SYS_HAS_CPU_RM7000
 	select SYS_HAS_CPU_NEVADA
 	select SYS_SUPPORTS_64BIT_KERNEL
--- linux-6.6/arch/mips/sgi-ip32/Platform.vanilla	2024-01-06 20:56:28.524478690 +0100
+++ linux-6.6/arch/mips/sgi-ip32/Platform	2024-01-06 20:57:21.112484501 +0100
@@ -7,4 +7,5 @@
 # will break.
 #
 cflags-$(CONFIG_SGI_IP32)	+= -I$(srctree)/arch/mips/include/asm/mach-ip32
+cflags-$(CONFIG_CPU_R10000)	+= -mr10k-cache-barrier=load-store
 load-$(CONFIG_SGI_IP32)		+= 0xffffffff80004000
